AutofacContrib.NSubstitute/SubstituteForBuilder.cs

```diff
@@ -28,6 +29,43 @@ public AutoSubstituteBuilder Configure(Action<TService> action)
             action(_service);
             return _builder;
         }
-    }
 
+        /// <summary>
+        /// Allows for configuration of the service with access to the resolved components.
+        /// </summary>
+        /// <param name="action">The delegate to configure the service.</param>
+        /// <returns>The original <see cref="AutoSubstituteBuilder"/>.</returns>
+        public AutoSubstituteBuilder Configure(Action<TService, IComponentContext> action)
+        {
+            _builder.ConfigureBuilder(builder =>
+            {
+                builder.RegisterBuildCallback(scope =>
+                {
+                    action(_service, scope);
+                });
+            });
+
+            return _builder;
+        }
+
+        /// <summary>
+        /// Completes the configuration of the substitute.
+        /// </summary>
+        /// <returns>The original <see cref="AutoSubstituteBuilder"/>.</returns>
+        public AutoSubstituteBuilder Configured()
+        {
+            return _builder;
+        }
+
+        /// <summary>
```

Comment:
These xmldoc comments aren't helping me understand it either so probably need a tweak :)

Reply:
Now?